mirrulations-work-server/src/mirrserver/put_results_validator.py

```diff
@@ -0,0 +1,46 @@
+class InvalidResultsException(Exception):
+    """
+    Raised when no results/json data are sent
+    """
+    message = {'error': 'The body does not contain the results'}
+    status_code = 403
+
+
+class InvalidClientIDException(Exception):
+    """
+    Raised when the client ID is invalid
+    (not 0<=client_id<20 or not an integer)
+    """
+    message = {'error': 'Invalid client ID'}
+    status_code = 401
+
+
+class MissingClientIDException(Exception):
+    """
+    Raised when the client ID is missing
+    """
+    message = {'error': 'Client ID was not provided'}
+    status_code = 401
+
+
+class JobCompleted(Exception):
+    """
+    Raised when when the job successfully completed
+    """
+    message = {'success': 'Job was successfully completed'}
+    status_code = 200
+
```

Comment:
We probably should not be raising an exception when the job is completed successfully, an exception is supposed to be an error that disrupts the normal flow of the program's instructions. A success is the opposite and is the intended functionality.

Reply:
my reasoning behind it is that success or job completed is one of responses we should get from the PutResultsValidator instead of returning that in the work server. I get that Jobcompleted() shouldn't be an exception, however,  it should be handled on the PutResultsValidator class where check_put_results(self, data, client_id): return the message and the code. I'll try to modify the code then pushed it again. thanks for your review